vdom/patch-op.js

```diff
@@ -120,20 +120,25 @@ function destroyWidget(domNode, w) {
 
 function reorderChildren(domNode, moves) {
     var childNodes = domNode.childNodes
-
-    for (var i = 0; i < moves.length; i++) {
-        var move = moves[i]
-        var from = move.from;
-        var to = move.to;
-
-        if (to >= 0) {
-            domNode.insertBefore(
-                childNodes[move.from],
-                childNodes[move.to]
-            )
-        } else {
-            domNode.removeChild(childNodes[from])
+    var keyMap = {}
+    var node
+    var remove
+    var insert
+
+    for (var i = 0; i < moves.removes.length; i++) {
+        remove = moves.removes[i]
+        node = childNodes[remove.from]
+        if (remove.key) {
+            keyMap[remove.key] = node
         }
+        domNode.removeChild(node)
+    }
+
+    for (var j = 0; j < moves.inserts.length; j++) {
+        insert = moves.inserts[j]
+        node = keyMap[insert.key]
+        // this is the weirdest bug i've ever seen in webkit
```

Comment:
it's probably undefined behavior. Maybe there is a better way than checking the child node length? Don't really want to be checking a host object property in a tight loop when we knew upfront that it was an append operation.